### ISSUE TITLE
TLS setting problem

### ISSUE THREAD

**Issue by jys923** [AirenSoft/OvenMediaEngine#144]
I want setting TLS in OME

err Msg

`
yoon@yoon-System-Product-Name:~$ systemctl status ovenmediaengine.service
● ovenmediaengine.service - OvenMediaEngine
   Loaded: loaded (/lib/systemd/system/ovenmediaengine.service; linked; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2020-07-14 12:42:10 KST; 5s ago
  Process: 26871 ExecStart=/usr/bin/OvenMediaEngine -d (code=exited, status=1/FAILURE)
 Main PID: 20632 (code=killed, signal=TERM)

 7월 14 12:42:10 yoon-System-Product-Name systemd[1]: Starting OvenMediaEngine...
 7월 14 12:42:10 yoon-System-Product-Name OvenMediaEngine[26871]: [2020-07-14 12:42:10.042] E 26872 Config | item.cpp:346  | Unknown configuration found: <Server>.<Bind>.<Publishers>.<WebRTC>.<Signalling>
 7월 14 12:42:10 yoon-System-Product-Name OvenMediaEngine[26871]: [2020-07-14 12:42:10.042] E 26872 OvenMediaEngine | main.cpp:313  | An error occurred while load config
 7월 14 12:42:10 yoon-System-Product-Name OvenMediaEngine[26871]: [2020-07-14 12:42:10.043] E 26871 OvenMediaEngine | main.cpp:298  | An error occurred while creating daemon
 7월 14 12:42:10 yoon-System-Product-Name systemd[1]: ovenmediaengine.service: Control process exited, code=exited status=1
 7월 14 12:42:10 yoon-System-Product-Name systemd[1]: ovenmediaengine.service: Failed with result 'exit-code'.
 7월 14 12:42:10 yoon-System-Product-Name systemd[1]: Failed to start OvenMediaEngine.

`

user certificate made by letsencrypt in nginx certbot

my server.xml
'
<?xml version="1.0" encoding="UTF-8" ?>
<Server version="7">
	<Name>OvenMediaEngine</Name>
	
	<Type>origin</Type>
	
	<IP>myIP</IP>

	
	<Bind>
		<Providers>
			<RTMP>
				<Port>1935</Port>
			</RTMP>
			<MPEGTS>
				
				<Port>4000-4005/udp</Port>
			</MPEGTS>
		</Providers>

		<Publishers>
			<OVT>
				<Port>9000</Port>
			</OVT>
			<HLS>
				<Port>80</Port>
				
				<TLSPort>443</TLSPort>
			</HLS>
			<DASH>
				<Port>80</Port>
				
				<TLSPort>443</TLSPort>
			</DASH>
			<WebRTC>
				<Signalling>
					<Port>3333</Port>
					
					<TLSPort>3334</TLSPort>
				</Signalling>
				<IceCandidates>
					<IceCandidate>*:10000-10005/udp</IceCandidate>
				</IceCandidates>
			</WebRTC>
		</Publishers>
	</Bind>

	
	

	<VirtualHosts>
		
		<VirtualHost include="VHost*.xml" />
		<VirtualHost>
			<Name>default</Name>

			
			<Domain>
				<Names>
					
					<Name>mydomain.com</Name>
				</Names>
				
				<TLS>
					<CertPath>/etc/letsencrypt/live/mydomain.com/cert.pem</CertPath>
					<KeyPath>/etc/letsencrypt/live/mydomain.com/privkey.pem</KeyPath>
					<ChainCertPath>/etc/letsencrypt/live/mydomain.com/fullchain.pem</ChainCertPath>
				</TLS>
				
			</Domain>
			
			
			<Applications>
				<Application>
					<Name>app</Name>
					
					<Type>live</Type>
					<Encodes>
						<Encode>
							<Name>HD_H264_AAC</Name>
							<Audio>
								<Codec>aac</Codec>
								<Bitrate>128000</Bitrate>
								<Samplerate>48000</Samplerate>
								<Channel>2</Channel>
							</Audio>
							<Video>
								<Codec>h264</Codec>
								<Width>1280</Width>
								<Height>720</Height>
								<Bitrate>2000000</Bitrate>
								<Framerate>30.0</Framerate>
							</Video>
						</Encode>					
						<Encode>
							<Name>bypass</Name>
							<Audio>
								<Bypass>true</Bypass>
							</Audio>
							<Video>
								<Bypass>true</Bypass>
							</Video>
						</Encode>
						<Encode>
							<Name>opus</Name>
							<Audio>
								<Codec>opus</Codec>
								<Bitrate>128000</Bitrate>
								<Samplerate>48000</Samplerate>
								<Channel>2</Channel>
							</Audio>
						</Encode>
					</Encodes>
					<Streams>
						<Stream>
							<Name>${OriginStreamName}</Name>
							<Profiles>
								<Profile>HD_H264_AAC</Profile>
							</Profiles>
						</Stream>					
						<Stream>
							<Name>${OriginStreamName}</Name>
							<Profiles>
								<Profile>bypass</Profile>
								<Profile>opus</Profile>
							</Profiles>
						</Stream>
					</Streams>
					<Providers>
						<OVT />
						<RTMP />
						<MPEGTS>
							<StreamMap>
								
								<Stream>
									<Name>stream_${Port}</Name>
									<Port>4000,4001-4004</Port>
								</Stream>
								<Stream>
									<Name>stream_4005</Name>
									<Port>4005</Port>
								</Stream>
							</StreamMap>
						</MPEGTS>
						<RTSPPull />
					</Providers>
					<Publishers>
						<ThreadCount>4</ThreadCount>
						<OVT />
						<WebRTC>
							<Timeout>30000</Timeout>
						</WebRTC>
						<HLS>
							<SegmentDuration>5</SegmentDuration>
							<SegmentCount>3</SegmentCount>
							<CrossDomain>
								<Url>*</Url>
							</CrossDomain>
						</HLS>
						<DASH>
							<SegmentDuration>5</SegmentDuration>
							<SegmentCount>3</SegmentCount>
							<CrossDomain>
								<Url>*</Url>
							</CrossDomain>
						</DASH>
						<LLDASH>
							<SegmentDuration>5</SegmentDuration>
							<CrossDomain>
								<Url>*</Url>
							</CrossDomain>
						</LLDASH>
					</Publishers>
				</Application>
			</Applications>
		</VirtualHost>
	</VirtualHosts>
</Server>
'

**Comment by dimiden** [comment]
@jys923 
1. Since the log below was printed, I think there is a wrong setting. Please attach the entire ```Server.xml``` and unmodified ```/var/log/ovenmediaengine/ovenmediaengine.log``` to determine the exact cause.
```[2020-07-14 12:42:10.042] E 26872 Config | item.cpp:346 | Unknown configuration found: ....```

2. OME doesn't support ```.pem``` files. Therefore, you have to convert them to ```.crt```/```.key``` using the following command:

- Convert ```.pem``` to ```.key```
```
openssl rsa -in /etc/letsencrypt/live/mydomain.com/privkey.pem -text > privkey.key
```
- Convert ```.pem``` to ```.crt```
```
openssl x509 -inform PEM -in /etc/letsencrypt/live/mydomain.com/cert.pem -out cert.crt
openssl x509 -inform PEM -in /etc/letsencrypt/live/mydomain.com/fullchain.pem -out fullchain.crt
```

**Comment by jys923** [comment]
thank ypu for your answer

`[2020-07-14 14:05:44.133] I 29471 Config | config_manager.cpp:144  | Trying to set logfile in directory... (/var/log/ovenmediaengine)
[2020-07-14 14:05:44.134] I 29471 Config | config_manager.cpp:47   | Trying to load configurations... (/usr/share/ovenmediaengine/conf/Server.xml)
[2020-07-14 14:05:44.134] E 29471 Config | item.cpp:346  | Unknown configuration found: <Server>.<Bind>.<Publishers>.<WebRTC>.<Signalling>.<TLSPort>
[2020-07-14 14:05:44.134] E 29471 OvenMediaEngine | main.cpp:313  | An error occurred while load config`

![주석 2020-07-14 141736](https://user-images.githubusercontent.com/18624102/87387013-cbbf2480-c5dc-11ea-9d3e-942593909a75.png)

`
[2020-07-14 14:19:07.718] I 30691 Config | config_manager.cpp:144  | Trying to set logfile in directory... (/var/log/ovenmediaengine)
[2020-07-14 14:19:07.718] I 30691 Config | config_manager.cpp:47   | Trying to load configurations... (/usr/share/ovenmediaengine/conf/Server.xml)
[2020-07-14 14:19:07.719] E 30691 Config | item.cpp:346  | Unknown configuration found: <Server>.<Bind>.<Publishers>.<WebRTC>.<Signalling>.<TLSPort>
[2020-07-14 14:19:07.719] E 30691 OvenMediaEngine | main.cpp:313  | An error occurred while load config
[2020-07-14 14:19:55.637] I 30767 Config | config_manager.cpp:144  | Trying to set logfile in directory... (/var/log/ovenmediaengine)
[2020-07-14 14:19:55.638] I 30767 Config | config_manager.cpp:47   | Trying to load configurations... (/usr/share/ovenmediaengine/conf/Server.xml)
[2020-07-14 14:19:55.638] E 30767 Config | item.cpp:346  | Unknown configuration found: <Server>.<Bind>.<Publishers>.<WebRTC>.<Signalling>.<TLSPort>
[2020-07-14 14:19:55.638] E 30767 OvenMediaEngine | main.cpp:313  | An error occurred while load config
`

![주석 2020-07-14 142258](https://user-images.githubusercontent.com/18624102/87387376-86e7bd80-c5dd-11ea-857d-a8c3230bda61.png)

----------------------
<?xml version="1.0" encoding="UTF-8" ?>

<Server version="7">
	<Name>OvenMediaEngine</Name>
	
	<Type>origin</Type>
	
	<IP>*</IP>

	
	<Bind>
		<Providers>
			<RTMP>
				<Port>1935</Port>
			</RTMP>
			<MPEGTS>
				
				<Port>4000-4005/udp</Port>
			</MPEGTS>
		</Providers>

		<Publishers>
			<OVT>
				<Port>9000</Port>
			</OVT>
			<HLS>
				<Port>80</Port>
				
				<TLSPort>443</TLSPort>
			</HLS>
			<DASH>
				<Port>80</Port>
				
				<TLSPort>443</TLSPort>
			</DASH>
			<WebRTC>
				<Signalling>
					<Port>3333</Port>
					
					<TLSPort>3334</TLSPort>
				</Signalling>
				<IceCandidates>
					<IceCandidate>*:10000-10005/udp</IceCandidate>
				</IceCandidates>
			</WebRTC>
		</Publishers>
	</Bind>

	
	

	<VirtualHosts>
		
		<VirtualHost include="VHost*.xml" />
		<VirtualHost>
			<Name>default</Name>

			
			<Domain>
				<Names>
					
					<Name>mydomain.com</Name>
				</Names>
				
				<TLS>
					<CertPath>/etc/letsencrypt/live/mydomain.com/cert.crt</CertPath>
					<KeyPath>/etc/letsencrypt/live/mydomain.com/privkey.key</KeyPath>
					<ChainCertPath>/etc/letsencrypt/live/mydomain.com/chain.crt</ChainCertPath>
				</TLS>
				
			</Domain>
			
			
			<Applications>
				<Application>
					<Name>app</Name>
					
					<Type>live</Type>
					<Encodes>
						<Encode>
							<Name>HD_H264_AAC</Name>
							<Audio>
								<Codec>aac</Codec>
								<Bitrate>128000</Bitrate>
								<Samplerate>48000</Samplerate>
								<Channel>2</Channel>
							</Audio>
							<Video>
								<Codec>h264</Codec>
								<Width>1280</Width>
								<Height>720</Height>
								<Bitrate>2000000</Bitrate>
								<Framerate>30.0</Framerate>
							</Video>
						</Encode>					
						<Encode>
							<Name>bypass</Name>
							<Audio>
								<Bypass>true</Bypass>
							</Audio>
							<Video>
								<Bypass>true</Bypass>
							</Video>
						</Encode>
						<Encode>
							<Name>opus</Name>
							<Audio>
								<Codec>opus</Codec>
								<Bitrate>128000</Bitrate>
								<Samplerate>48000</Samplerate>
								<Channel>2</Channel>
							</Audio>
						</Encode>
					</Encodes>
					<Streams>
						<Stream>
							<Name>${OriginStreamName}</Name>
							<Profiles>
								<Profile>HD_H264_AAC</Profile>
							</Profiles>
						</Stream>					
						<Stream>
							<Name>${OriginStreamName}</Name>
							<Profiles>
								<Profile>bypass</Profile>
								<Profile>opus</Profile>
							</Profiles>
						</Stream>
					</Streams>
					<Providers>
						<OVT />
						<RTMP />
						<MPEGTS>
							<StreamMap>
								
								<Stream>
									<Name>stream_${Port}</Name>
									<Port>4000,4001-4004</Port>
								</Stream>
								<Stream>
									<Name>stream_4005</Name>
									<Port>4005</Port>
								</Stream>
							</StreamMap>
						</MPEGTS>
						<RTSPPull />
					</Providers>
					<Publishers>
						<ThreadCount>4</ThreadCount>
						<OVT />
						<WebRTC>
							<Timeout>30000</Timeout>
						</WebRTC>
						<HLS>
							<SegmentDuration>5</SegmentDuration>
							<SegmentCount>3</SegmentCount>
							<CrossDomain>
								<Url>*</Url>
							</CrossDomain>
						</HLS>
						<DASH>
							<SegmentDuration>5</SegmentDuration>
							<SegmentCount>3</SegmentCount>
							<CrossDomain>
								<Url>*</Url>
							</CrossDomain>
						</DASH>
						<LLDASH>
							<SegmentDuration>5</SegmentDuration>
							<CrossDomain>
								<Url>*</Url>
							</CrossDomain>
						</LLDASH>
					</Publishers>
				</Application>
			</Applications>
		</VirtualHost>
	</VirtualHosts>
</Server>

**Comment by dimiden** [comment]
@jys923 
We apologize for the confusion. There was a typo recently modifying the ```Server.xml``` structure.
You can do this in one of two ways:

1. Build and use the latest commit (https://github.com/AirenSoft/OvenMediaEngine/commit/1fb3d5faa89df17a05d198030eb2635f02b3a999)
2. Temporarily change the name of the ```<TLSPort>``` in ```<Signalling>``` to ```<TlsPort>```. (Note the case)
```
AS-IS
<Bind>
	<Publishers>
		<WebRTC>
			<Signalling>
				<TLSPort>3334</TLSPort> <=== Change here
-----
TO-BE
<Bind>
	<Publishers>
		<WebRTC>
			<Signalling>
				<TlsPort>3334</TlsPort> <=== Like this
```

**Comment by jys923** [comment]
well done
thank you for your kind answer.

**Comment by jys923** [comment]
well done
thank you for your kind answer.

**Comment by getroot** [comment]
This has been fixed in version 0.10.5.
